analytics/tests/test_views.py

```diff
@@ -789,41 +792,44 @@ def check_realm_reactivation_link_query_result(result: HttpResponse) -> None:
         check_zulip_realm_query_result(result)
         check_lear_realm_query_result(result)
 
-        self.client_post("/accounts/home/", {"email": self.nonreg_email("test")})
-        self.login("iago")
-        result = self.client_get("/activity/support", {"q": self.nonreg_email("test")})
-        check_preregistration_user_query_result(result, self.nonreg_email("test"))
-        check_zulip_realm_query_result(result)
+        with mock.patch(
+            "analytics.views.timezone_now", return_value=timezone_now() - timedelta(minutes=50)
+        ):
+            self.client_post("/accounts/home/", {"email": self.nonreg_email("test")})
+            self.login("iago")
+            result = self.client_get("/activity/support", {"q": self.nonreg_email("test")})
+            check_preregistration_user_query_result(result, self.nonreg_email("test"))
+            check_zulip_realm_query_result(result)
 
-        stream_ids = [self.get_stream_id("Denmark")]
-        invitee_emails = [self.nonreg_email("test1")]
-        self.client_post(
-            "/json/invites",
-            {
-                "invitee_emails": invitee_emails,
-                "stream_ids": orjson.dumps(stream_ids).decode(),
-                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
-            },
-        )
-        result = self.client_get("/activity/support", {"q": self.nonreg_email("test1")})
-        check_preregistration_user_query_result(result, self.nonreg_email("test1"), invite=True)
-        check_zulip_realm_query_result(result)
+            stream_ids = [self.get_stream_id("Denmark")]
+            invitee_emails = [self.nonreg_email("test1")]
+            self.client_post(
+                "/json/invites",
+                {
+                    "invitee_emails": invitee_emails,
+                    "stream_ids": orjson.dumps(stream_ids).decode(),
+                    "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                },
+            )
+            result = self.client_get("/activity/support", {"q": self.nonreg_email("test1")})
+            check_preregistration_user_query_result(result, self.nonreg_email("test1"), invite=True)
+            check_zulip_realm_query_result(result)
 
-        email = self.nonreg_email("alice")
-        self.client_post("/new/", {"email": email})
-        result = self.client_get("/activity/support", {"q": email})
-        check_realm_creation_query_result(result, email)
+            email = self.nonreg_email("alice")
+            self.client_post("/new/", {"email": email})
+            result = self.client_get("/activity/support", {"q": email})
```

Comment:
@timabbott Do you think I should mock `timezone_now` for each `activity/support` request separately even if they'd be duplicates?

Reply:
I don't think it's necessary.